### PR TITLE
Replace `ServeStream` with `ServeContent`

### PR DESCRIPTION
**Pull request by KN4CK3R**
PR from Discord

Replace `ServeStream` with `ServeContent` which enabled HTTP range support for downloads. Only the package registry and the repo archiver use `ServeStream`.

A possible downside may be that `http.ServeContent` uses the `Seek` method to get the length of the stream. Sadly there is no way to pass the size as parameter even if we know it.

### CODE REVIEW

File: modules/timeutil/timestamp.go [hunk 1]
```diff
@@ -54,6 +54,12 @@ func (ts TimeStamp) AsTime() (tm time.Time) {
 	return ts.AsTimeInLocation(setting.DefaultUILocation)
 }
 
+// AsLocalTime convert timestamp as time.Time in local location
+func (ts TimeStamp) AsLocalTime() (tm time.Time) {
+	tm = time.Unix(int64(ts), 0)
+	return tm
```
Comment (delvh) [inline modules/timeutil/timestamp.go]:
```suggestion
func (ts TimeStamp) AsLocalTime() time.Time {
	return time.Unix(int64(ts), 0)
```

**Comment by zeripath** [comment]
I guess the question is if we think this is a significant slowdown because of this?

A possibility might be to use ServeFile to somehow avoid it if necessary.

**Review by zeripath — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Comment by KN4CK3R** [comment]
> I guess the question is if we think this is a significant slowdown because of this?

Yes. The best solution is if https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/net/http/fs.go;drc=8d57f4dcef5d69a0a3f807afaa9625018569010b;bpv=1;bpt=1;l=221 would be exported. Then we could simply pass the content size and the seek would just occur if it is needed because of the http range header. It's just unnecessary work at the moment.

**Comment by zeripath** [comment]
I wonder if we can instead refactor to use ServeFile instead but I'm not certain how worth it it would be

**Review by lunny — APPROVED** [review]

**Pull request merged**